src/store/TimeperiodStore.js

```diff
@@ -119,7 +119,9 @@ export default class TimeperiodStore {
         var stx = this.context.stx;
         var dataSet = stx.chart.dataSet;
         var price = dataSet[dataSet.length-1].Close;
-        return stx.pixelFromPrice(price, stx.chart.panel) + 18;
+        var x = stx.pixelFromPrice(price, stx.chart.panel);
+        x = x > stx.chart.panel.bottom - 60 ? x - 18 : x + 18;
```

Comment:
it's a nitpick, but I'd prefer you use `let` and `const` over `var` whenever possible.

Also (this one is not a nitpick) do not to use magic numbers (numbers that appear without explanation). Either add a comment explaining or use constants:

```js
x += (x > stx.chart.panel.bottom - WHAT_IS_THIS) ? -PROPER_NAME : +PROPER_NAME ;
```